catalogue/webapp/pages/work.tsx

```diff
@@ -1,71 +1,65 @@
-import { NextPage, NextPageContext } from 'next';
-import Router from 'next/router';
+import { GetServerSideProps, NextPage } from 'next';
 import {
   Work as WorkType,
   CatalogueApiError,
 } from '@weco/common/model/catalogue';
-import { workLink } from '@weco/common/services/catalogue/routes';
 import ErrorPage from '@weco/common/views/components/ErrorPage/ErrorPage';
 import Work from '../components/Work/Work';
 import { getWork } from '../services/catalogue/works';
+import {
+  getGlobalContextData,
+  WithGlobalContextData,
+} from '@weco/common/views/components/GlobalContextProvider/GlobalContextProvider';
 
 type Props = {
   workResponse: WorkType | CatalogueApiError;
-};
-
-export const WorkPage: NextPage<Props> = ({ workResponse }) => {
-  if (
-    workResponse.type === 'Work' ||
-    workResponse.type === 'Collection' ||
-    workResponse.type === 'Section' ||
-    workResponse.type === 'Series'
-  ) {
-    return <Work work={workResponse} />;
-  }
+} & WithGlobalContextData;
 
-  if (workResponse.type === 'Error') {
-    return (
-      <ErrorPage
-        statusCode={workResponse.httpStatus}
-        title={workResponse.description}
-      />
-    );
-  }
+export const WorkPage: NextPage<Props> = ({
+  workResponse,
+  globalContextData,
+}: Props) => {
+  return (
+    <>
+      {workResponse.type === 'Error' && (
+        <ErrorPage
+          statusCode={workResponse.httpStatus}
+          title={workResponse.description}
+          globalContextData={globalContextData}
+        />
+      )}
+      {workResponse.type !== 'Error' && (
+        <Work work={workResponse} globalContextData={globalContextData} />
+      )}
+    </>
+  );
 };
 
-WorkPage.getInitialProps = async (ctx: NextPageContext) => {
-  const { id } = ctx.query;
+export const getServerSideProps: GetServerSideProps<Props> = async context => {
+  const globalContextData = getGlobalContextData(context);
+  const { id } = context.query;
 
   const workResponse = await getWork({
     id,
-    toggles: ctx.query.toggles,
+    toggles: globalContextData.toggles,
   });
 
-  if (workResponse) {
-    if (workResponse.type === 'Redirect') {
-      const { res } = ctx;
-      if (res) {
-        res.writeHead(workResponse.status, {
-          Location: workResponse.redirectToId,
-        });
-        res.end();
-      } else {
-        const link = workLink({ id: workResponse.redirectToId });
-        Router.push(link.href, link.as);
-      }
-    }
-
+  if (workResponse.type === 'Redirect') {
     return {
-      workResponse:
-        workResponse.type === 'Work' ||
-        workResponse.type === 'Error' ||
-        workResponse.type === 'Collection' ||
-        workResponse.type === 'Section' ||
-        workResponse.type === 'Series'
-          ? workResponse
-          : undefined,
+      props: {},
```

Comment:
I'm having trouble seeing what happens if this is what is returned

Reply:
You need to have it here for the type system. We don't send props to the redirect, it just redirects.